HW1/P4/P4.py

```diff
@@ -0,0 +1,47 @@
+from P4_bfs import *
+
+import itertools
+
+# Initialize Spark
+from pyspark import SparkContext
+sc = SparkContext()
+sc.setLogLevel("ERROR")
+
+# Helper function used to clean data loaded from CSV file
+# Splits each line by comma, with error-trapping for names with commas
+# Removes "" around names and strips out whitespace
+def split_csv(val):
+    split = val.split(',')
+    name = ','.join(split[:-1]).replace('"','').strip()
+    comic = split[-1].replace('"','')
+    return tuple([comic, name])
+
+# Helper function calculates all 2-character permutations for a comic book
+def perm_chars(val):
+    return list(itertools.permutations(val[1], 2))
```

Comment:
Very cool, I didn't think of using itertools. It is a compact way to deal with links in our networks.